### PR TITLE
feat(machine): support newer versions of Ubuntu

### PR DESCRIPTION
**Pull request by jaredallard**
Adds support for newer versions of Ubuntu and Debian by using the `yq`
apt package, if it's available, and falling back to `pip3` if it's not.

Cleans up the code a bit to make it easier to determine what version of
`gh` is in use/desired. Updates to `v2.32.1`. This will be used by
`bootstrap/dev` in an upcoming PR.

### CODE REVIEW

File: shell/circleci/machine.sh [hunk 1]
```diff
@@ -21,40 +32,17 @@ fi
 # Add APT repositories we do need.
 if should_install_vault; then
   wget -O- https://apt.releases.hashicorp.com/gpg | gpg --dearmor -o hashicorp-archive-keyring.gpg
-
-  # 2023-02-22: Hashicorp updated their keyring but removed their older key, this reulted in no
```
Comment (jaredallard) [inline shell/circleci/machine.sh:25]: Removed this since the parent issue has been closed/resolved.

**Comment by malept** [comment]
Doesn't this repo control which CircleCI executor is used? Or is supporting older Debian/Ubuntu versions because it's configurable in the repo's generated ci.yml?

**Comment by jaredallard** [comment]
> Doesn't this repo control which CircleCI executor is used? Or is supporting older Debian/Ubuntu versions because it's configurable in the repo's generated ci.yml?

It generally does! However, it can be configured. Also, it's used in some Docker images to handle bootstrapping the image (see https://github.com/getoutreach/outreach-docker/pull/682)

File: shell/circleci/machine.sh [hunk 2]
```diff
@@ -21,52 +32,41 @@ fi
...
+
+  echo "Failed to install yq through apt, falling back to pip3 install (this only works on Ubuntu 22.04 and below)"
+  sudo pip3 install yq
+fi
 
-# Vault
 if should_install_vault; then
   echo "Installing Vault"
   sudo apt-get install -y vault
+  sudo rm -rf /opt/vault
```
Comment (malept) [inline shell/circleci/machine.sh:71]: Question: what does this delete?

**Review by malept — APPROVED** [review]
This seems fine, just one question.

File: shell/circleci/machine.sh (hunk 2 above)
Comment (jaredallard) [reply shell/circleci/machine.sh:71]: Ah, this just cleans up the vault server TLS certs and whatnot that were already generated. Seemed better than leaving it around.

**Pull request merged**